src/react.ts

```diff
@@ -47,10 +49,15 @@ export function useStore<TState, StateSlice>(
   selector: (state: TState) => StateSlice = api.getState as any,
   equalityFn?: (a: StateSlice, b: StateSlice) => boolean
 ) {
-  if (import.meta.env?.MODE !== 'production' && equalityFn) {
+  if (
+    import.meta.env?.MODE !== 'production' &&
+    equalityFn &&
+    !didWarnAboutEqualityFn
+  ) {
     console.warn(
-      "[DEPRECATED] Use `createWithEqualityFn` from 'zustand/traditional'. https://github.com/pmndrs/zustand/discussions/1937"
+      "[DEPRECATED] Use `createWithEqualityFn` instead of `create` or use `useStoreWithEqualityFn` instead of `useStore`. They can be imported from 'zustand/traditional'. https://github.com/pmndrs/zustand/discussions/1937"
     )
+    didWarnAboutEqualityFn = true
   }
```

Comment:
It may be a good idea to extract this to `equalityFn && warnAboutEqualityFnOnce()`

Reply:
Unfortunately, that doesn't work with minification/dead-code-elimination in production.

Reply:
I see, I don't have any more suggestions then.